src/pages/seo.tsx

```diff
@@ -0,0 +1,99 @@
+import React from 'react';
+import { Helmet } from 'react-helmet';
+import { useStaticQuery, graphql } from 'gatsby';
+
+const SEO = ({ description = '', lang = 'en', meta = [], title }: SEOProps) => {
+  const { site } = useStaticQuery<QueryTypes>(SEOStaticQuery);
+
+  const metaDescription = description || site.siteMetadata.description;
+  const defaultTitle = site.siteMetadata?.title;
+
+  return (
+    <Helmet
+      htmlAttributes={{
+        lang,
+      }}
+      title={title}
+      titleTemplate={defaultTitle ? `%s | ${defaultTitle}` : undefined}
+      meta={[
+        {
+          name: `description`,
+          content: metaDescription,
+        },
+        {
+          property: `og:title`,
+          content: title,
+        },
+        {
+          property: `og:description`,
+          content: metaDescription,
+        },
+        {
+          property: `og:type`,
+          content: `website`,
+        },
+        {
+          name: `twitter:card`,
+          content: `summary`,
+        },
+        {
+          name: `twitter:creator`,
+          content: site.siteMetadata?.author || ``,
+        },
+        {
+          name: `twitter:title`,
+          content: title,
+        },
+        {
+          name: `twitter:description`,
+          content: metaDescription,
+        },
+      ].concat(meta)}
+    />
+  );
+};
+
+export default SEO;
+
+// Types
+type SEOProps = {
+  description?: string;
+  lang?: string;
+  meta?: Meta;
+  title: string;
+};
+
+type Meta = ConcatArray<PropertyMetaObj | NameMetaObj>;
+
+type PropertyMetaObj = {
```

Comment:
+1 on being an interface and split out; same for line 73 `NameMetaObj` and line 78 `QueryTypes`